doc/api/server.md

```diff
@@ -0,0 +1,69 @@
+# Server
+
+## Class: jstp.Server
+
+JSTP server base class with necessary methods.
+All of the transport servers extend this class.
+
+### Event: 'connect'
+
+- [`<Connection>`][connection]
+
+Emitted when a JSTP connection is established and new session is created.
+
+### Event: 'reconnect'
+
+- [`<Connection>`][connection]
+
+Emitted when a JSTP connection is established that is reusing existing
+[session][session].
+
+### Event: 'disconnect'
+
+- [`<Connection>`][connection]
+
+Emitted when a JSTP connection is closed.
+
+### server.getClients()
+
+- Returns: [`<Iterator>`][iterator]
+
+Get all clients as an Iterator of [`Connection`][connection] instances.
+
+### server.getClientsArray()
+
+- Returns: [`<Connection[]>`][connection]
+
+### server.updateApplications(applications)
+
+- `applications` [`<Application[]>`][application] | [`Map`][map] New
+  applications array or index obtained from
+  [`jstp.createAppsIndex()`][createappindex].
+
+Replaces set of applications/applications versions available when connecting to
+the server.
+
+### server.updateConnectionsApi()
+
+This function updates application version for each connection.
+
+It will update connection's application to the newest version in the range
+requested by the client. For example, if the client requested `'1.1.x'` range,
+it will not be updated to `'1.2.x'`. If no suitable version is found, the client
```

Comment:
For example, if the client requested `'1.1.x'` range and was using `'1.1.1'` version, it will be updated to the latest SemVer compatible version in this range (e.g. `'1.1.2'` but not `'1.2.x'`).